Touch Bar Simulator/Glue.swift

```diff
@@ -1,6 +1,24 @@
 import Foundation
 import Defaults
 
+extension Defaults {
+	@discardableResult
+	func observe<T: Codable, Weak: AnyObject>(_ key: Key<T>, tiedToLifetimeOf weaklyHeldObject: Weak, options: NSKeyValueObservingOptions = [.initial, .new, .old], handler: @escaping (KeyChange<T>) -> Void) -> DefaultsObservation {
+		var observation: DefaultsObservation!
+		observation = observe(key, options: options) { [weak weaklyHeldObject] change in
+			guard let temporaryStrongReference = weaklyHeldObject else {
+				// Will never occur on first call (outer function holds a strong reference),
+				// so observation will never be nil
+				observation.invalidate()
+				return
+			}
+			_ = temporaryStrongReference
+			handler(change)
+		}
+		return observation
+	}
+}
```

Comment:
Would be awesome if you could submit this to https://github.com/sindresorhus/Defaults 🙏